test/e2e/set/scyllacluster/scyllacluster_scaling.go

```diff
@@ -211,4 +216,65 @@ var _ = g.Describe("ScyllaCluster", func() {
 		verifyCQLData(ctx, diRF2)
 		verifyCQLData(ctx, diRF3)
 	})
+
+	g.It("should support vertical scaling", func() {
+		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+		defer cancel()
+
+		const membersCount = 3
+		sc := scyllafixture.BasicScyllaCluster.ReadOrFail()
+		o.Expect(sc.Spec.Datacenter.Racks).To(o.HaveLen(1))
+		sc.Spec.Datacenter.Racks[0].Members = membersCount
+		o.Expect(sc.Spec.Datacenter.Racks[0].Resources.Limits).NotTo(o.BeNil())
+		sc.Spec.Datacenter.Racks[0].Resources.Limits[corev1.ResourceCPU] = *resource.NewQuantity(1, resource.DecimalSI)
+
+		framework.By("Creating a ScyllaCluster with %d members using 1 cpu each", membersCount)
+		sc, err := f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Create(ctx, sc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaCluster to rollout (RV=%s)", sc.ResourceVersion)
+		waitCtx1, waitCtx1Cancel := utils.ContextForRollout(ctx, sc)
+		defer waitCtx1Cancel()
+		sc, err = utils.WaitForScyllaClusterState(waitCtx1, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, utils.WaitForStateOptions{}, utils.IsScyllaClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		verifyScyllaCluster(ctx, f.KubeClient(), sc)
+		hosts := getScyllaHostsAndWaitForFullQuorum(ctx, f.KubeClient().CoreV1(), sc)
+		o.Expect(hosts).To(o.HaveLen(membersCount))
+		di := insertAndVerifyCQLData(ctx, hosts)
+		defer di.Close()
+
+		framework.By("Raising CPU limits to 2 cpu each")
+		sc.ManagedFields = nil
+		sc.ResourceVersion = ""
+		sc.Spec.Datacenter.Racks[0].Resources.Limits[corev1.ResourceCPU] = *resource.NewQuantity(2, resource.DecimalSI)
+		scData, err := runtime.Encode(scheme.Codecs.LegacyCodec(scyllav1.GroupVersion), sc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		// TODO: Use generated Apply method when our clients have it.
+		sc, err = f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Patch(ctx, sc.Name, types.ApplyPatchType, scData, metav1.PatchOptions{
+			FieldManager: f.FieldManager(),
+			Force:        pointer.Bool(true),
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(sc.Spec.Datacenter.Racks).To(o.HaveLen(1))
+		o.Expect(sc.Spec.Datacenter.Racks[0].Resources.Limits).To(o.HaveKeyWithValue(
+			corev1.ResourceCPU,
+			resource.MustParse("2"),
+		))
+
+		framework.By("Waiting for the ScyllaCluster to rollout (RV=%s)", sc.ResourceVersion)
+		waitCtx2, waitCtx2Cancel := utils.ContextForRollout(ctx, sc)
+		defer waitCtx2Cancel()
+		sc, err = utils.WaitForScyllaClusterState(waitCtx2, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, utils.WaitForStateOptions{}, utils.IsScyllaClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		verifyScyllaCluster(ctx, f.KubeClient(), sc)
+
+		oldHosts := hosts
+		hosts = getScyllaHostsAndWaitForFullQuorum(ctx, f.KubeClient().CoreV1(), sc)
+		o.Expect(hosts).To(o.HaveLen(membersCount))
+		o.Expect(hosts).To(o.ConsistOf(oldHosts))
+
+		verifyCQLData(ctx, di)
```

Comment:
this is just testing whether resource change was propagated - we already have it in `scyllacluster_updates.go`. It should check whether Scylla picked it up, by checking number of shards before and after the change.